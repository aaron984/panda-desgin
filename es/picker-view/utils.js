export var setTransform = function setTransform(nodeStyle, value) {
  nodeStyle.transform = value;
  nodeStyle.webkitTransform = value;
};
export var setTransition = function setTransition(nodeStyle, value) {
  nodeStyle.transition = value;
  nodeStyle.webkitTransition = value;
};
export var Velocity = function () {
  var minInterval = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 30;
  var maxInterval = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 100;
  var time = 0;
  var initY = 0;
  var velocity = 0;
  var recorder = {
    record: function record(y) {
      var now = +new Date();
      velocity = (y - initY) / (now - time);

      if (now - time >= minInterval) {
        velocity = now - time <= maxInterval ? velocity : 0;
        initY = y;
        time = now;
      }
    },
    getVelocity: function getVelocity(y) {
      if (y !== initY) {
        recorder.record(y);
      }

      return velocity;
    }
  };
  return recorder;
}();