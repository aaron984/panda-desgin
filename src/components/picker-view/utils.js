export const setTransform = (nodeStyle, value) => {
  nodeStyle.transform = value;
  nodeStyle.webkitTransform = value;
};

export const setTransition = (nodeStyle, value) => {
  nodeStyle.transition = value;
  nodeStyle.webkitTransition = value;
};

export const Velocity = ((minInterval = 30, maxInterval = 100) => {
  let time = 0;
  let initY = 0;
  let velocity = 0;

  const recorder = {
    record: (y) => {
      const now = +new Date();
      velocity = (y - initY) / (now - time);

      if (now - time >= minInterval) {
        velocity = now - time <= maxInterval ? velocity : 0;
        initY = y;
        time = now;
      }
    },
    getVelocity: (y) => {
      if (y !== initY) {
        recorder.record(y);
      }
      return velocity;
    },
  };
  return recorder;
})();
