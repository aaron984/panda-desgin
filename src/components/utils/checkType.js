export const isObject = value => Object.prototype.toString.call(value) === '[object Object]';
export const isString = value => Object.prototype.toString.call(value) === '[object String]';
export const isNull = value => Object.prototype.toString.call(value) === '[object Null]';
export const isFunction = value => Object.prototype.toString.call(value) === '[object Function]';
export const isArray = value => Object.prototype.toString.call(value) === '[object Array]';
