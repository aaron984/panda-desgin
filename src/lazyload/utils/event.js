export function on(el, eventName, callback, opts = false) {
  el.addEventListener(eventName, callback, opts);
}

export function off(el, eventName, callback, opts = false) {
  opts = opts || false;
  el.removeEventListener(eventName, callback, opts);
}
