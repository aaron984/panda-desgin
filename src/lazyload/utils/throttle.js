/*eslint-disable */
export default function throttle(fn, threshhold = 250, scope) {
  let last;
  let deferTimer;

  return function () {
    let context = scope || this;

    let now = +new Date;
    let args = arguments;
    if (last && now < last + threshhold) {
      clearTimeout(deferTimer);
      deferTimer = setTimeout(function () {
        last = now;
        fn.apply(context, args);
      }, threshhold);
    } else {
      last = now;
      fn.apply(context, args);
    }
  };
}
