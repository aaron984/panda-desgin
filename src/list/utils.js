export default {
  getScrollEventTarget(element, rootParent = window) {
    let currentNode = element;
    while (currentNode && currentNode.tagName !== 'HTML' && currentNode.tagName !== 'BODY' && currentNode.nodeType === 1 && currentNode !== rootParent) {
      const overflowY = this.getComputedStyle(currentNode).overflowY;
      if (overflowY === 'scroll' || overflowY === 'auto') {
        return currentNode;
      }
      currentNode = currentNode.parentNode;
    }
    return rootParent;
  },

  getScrollTop(element) {
    return 'scrollTop' in element ? element.scrollTop : element.pageYOffset;
  },

  getElementTop(element) {
    return (element === window ? 0 : element.getBoundingClientRect().top) + this.getScrollTop(window);
  },

  getVisibleHeight(element) {
    return element === window ? element.innerHeight : element.getBoundingClientRect().height;
  },

  getComputedStyle: document.defaultView.getComputedStyle.bind(document.defaultView),
};
