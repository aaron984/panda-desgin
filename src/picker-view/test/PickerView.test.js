import React from 'react';
import { render, mount } from 'enzyme';
import { triggerDrag } from 'test/utils';
import PickerView from '../index';
import { Picker } from '../PickerView';

const getItems = () => {
  const items = [];
  for (let i = 0; i < 3; i++) {
    items.push((
      <PickerView.Item value={`${i}`} key={i}>
        count {i}
      </PickerView.Item>
    ));
  }
  return items;
};

describe('Pickerview', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('PickerView render correctly', () => {
    const props = {
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
    };
    const wrapper = render(<PickerView {...props}>{getItems()}</PickerView>);
    expect(wrapper).toMatchSnapshot();
  });

  it('PickerView set defaultSelectedValue should work fine', () => {
    const props = {
      defaultSelectedValue: '34',
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    expect(instance.state.selectedValue).toEqual('34');
  });

  it('PickerView set selectedValue should work fine', () => {
    const props = {
      selectedValue: '99',
      defaultSelectedValue: '23',
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    expect(instance.state.selectedValue).toEqual('99');
  });

  it('PickerView update selectedValue should work fine', () => {
    const props = {
      selectedValue: '99',
      defaultSelectedValue: '23',
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    wrapper.setProps({ selectedValue: '88', noAnimate: () => { } });
    wrapper.setProps({ selectedValue: '88' });
    expect(instance.state.selectedValue).toEqual('88');
  });

  it('PickerView if init props selectedValue is null update selectedValue should work fine', () => {
    const props = {
      defaultSelectedValue: '23',
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    wrapper.setProps({ a: '88' });
    wrapper.setProps({ selectedValue: '188' });
    expect(instance.state.selectedValue).toEqual('188');
  });

  it('PickerView default value should work fine', () => {
    const props = {
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    expect(instance.state.selectedValue).toEqual('0');

    wrapper.unmount();
  });

  it('PickerView getValue function should work fine', () => {
    const props = {
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    triggerDrag(instance.rootRef, 20, 0);
    jest.runAllTimers(300);
    expect(instance.getValue()).toEqual('0');
  });

  it('PickerView contral component getValue function should work fine', () => {
    const props = {
      selectedValue: '9',
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    expect(instance.getValue()).toEqual('9');
  });

  it('PickerView setDisabled function should work fine', () => {
    const props = {
      selectedValue: '9',
      onValueChange: jest.fn(),
      onScrollChange: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    instance.setDisabled(true);
    expect(instance.scrollDisabled).toEqual(true);
    expect(instance.onStart()).toEqual(undefined);
    expect(instance.onMove()).toEqual(undefined);
  });

  it('PickerView scrollingComplete function should work fine', () => {
    const props = {
      doScrollingComplete: jest.fn(),
      select: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    instance.scrollY = 5;
    instance.scrollingComplete();
    expect(props.doScrollingComplete).toHaveBeenCalled();
  });

  it('PickerView onScrollChange function should work fine', () => {
    const props = {
      doScrollingComplete: jest.fn(),
      select: jest.fn(),
      computeChildIndex: jest.fn(),
      onScrollChange: jest.fn(),
      noAnimate: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    instance.scrollY = 5;
    instance.scrollValue = 1;
    instance.onScrollChange();
    instance.scrollTo(0, 5);
    instance.scrollToWithoutAnimation(0, 5, 0);
  });

  it('PickerView fireValueChange function should work fine', () => {
    const props = {
      select: jest.fn(),
      onValueChange: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    instance.fireValueChange('32');
    expect(instance.state.selectedValue).toBe('32');
    expect(props.onValueChange).toHaveBeenCalled();
    instance.fireValueChange('32');
  });

  it('PickerView scrollHanders function should work fine', () => {
    const props = {
      select: jest.fn(),
      onValueChange: jest.fn(),
    };
    const wrapper = mount(<Picker {...props} >{getItems()}</Picker>);
    const instance = wrapper.instance();
    instance.scrollHanders().mousedown({ screenY: 2 });
    instance.scrollHanders().mousemove({ screenY: 2, preventDefault: () => { } });
    instance.scrollHanders().touchcancel({ screenY: 2 });
    instance.scrollHanders().mouseup({ screenY: 2 });
  });
});
