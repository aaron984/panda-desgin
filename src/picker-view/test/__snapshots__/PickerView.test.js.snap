// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pickerview PickerView render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "panda-picker-col",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "panda-picker-col__mask",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "panda-picker-col__indicator",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "panda-picker-col__content",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "panda-picker-col-item panda-picker-col-item-selected",
                },
                "children": Array [
                  Object {
                    "data": "count 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "panda-picker-col-item",
                  },
                  "children": Array [
                    Object {
                      "data": "count 1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "panda-picker-col-item",
                    },
                    "children": Array [
                      Object {
                        "data": "count 2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "panda-picker-col-item",
                },
                "children": Array [
                  Object {
                    "data": "count 1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "panda-picker-col-item",
                  },
                  "children": Array [
                    Object {
                      "data": "count 2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "panda-picker-col-item panda-picker-col-item-selected",
                  },
                  "children": Array [
                    Object {
                      "data": "count 0",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "panda-picker-col-item",
                },
                "children": Array [
                  Object {
                    "data": "count 2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "panda-picker-col-item",
                  },
                  "children": Array [
                    Object {
                      "data": "count 1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "panda-picker-col-item panda-picker-col-item-selected",
                    },
                    "children": Array [
                      Object {
                        "data": "count 0",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "panda-picker-col__indicator",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "panda-picker-col__content",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "panda-picker-col-item panda-picker-col-item-selected",
              },
              "children": Array [
                Object {
                  "data": "count 0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "panda-picker-col-item",
                },
                "children": Array [
                  Object {
                    "data": "count 1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "panda-picker-col-item",
                  },
                  "children": Array [
                    Object {
                      "data": "count 2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "panda-picker-col-item",
              },
              "children": Array [
                Object {
                  "data": "count 1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "panda-picker-col-item",
                },
                "children": Array [
                  Object {
                    "data": "count 2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "panda-picker-col-item panda-picker-col-item-selected",
                },
                "children": Array [
                  Object {
                    "data": "count 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "panda-picker-col-item",
              },
              "children": Array [
                Object {
                  "data": "count 2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "panda-picker-col-item",
                },
                "children": Array [
                  Object {
                    "data": "count 1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "panda-picker-col-item panda-picker-col-item-selected",
                  },
                  "children": Array [
                    Object {
                      "data": "count 0",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "panda-picker-col__mask",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "panda-picker-col__content",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "panda-picker-col-item panda-picker-col-item-selected",
            },
            "children": Array [
              Object {
                "data": "count 0",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "panda-picker-col-item",
              },
              "children": Array [
                Object {
                  "data": "count 1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "panda-picker-col-item",
                },
                "children": Array [
                  Object {
                    "data": "count 2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "panda-picker-col-item",
            },
            "children": Array [
              Object {
                "data": "count 1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "panda-picker-col-item",
              },
              "children": Array [
                Object {
                  "data": "count 2",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "panda-picker-col-item panda-picker-col-item-selected",
              },
              "children": Array [
                Object {
                  "data": "count 0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "panda-picker-col-item",
            },
            "children": Array [
              Object {
                "data": "count 2",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "panda-picker-col-item",
              },
              "children": Array [
                Object {
                  "data": "count 1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "panda-picker-col-item panda-picker-col-item-selected",
                },
                "children": Array [
                  Object {
                    "data": "count 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "panda-picker-col__indicator",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "panda-picker-col__mask",
            },
            "children": Array [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
